core/processor/processor.go

```diff
@@ -0,0 +1,122 @@
+package processor
+
+import (
+	"context"
+	"sync"
+
+	"chain/database/pg"
+	"chain/log"
+	"chain/protocol"
+	"chain/protocol/bc"
+)
+
+type CursorStore struct {
```

Comment:
The naming stuff we discussed offline: it would be cool if this could convey some of the notion that it's not safe to prune block data for the block the cursor is on or any subsequent block. (It may or may not be safe to prune earlier blocks.) I was thinking "pin" might work for that. So perhaps we would have types `pin.Store` and `pin.Pin`?